File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int> ans;
        for(const auto &query : queries) {
            int from = query[0];
            int to = query[1];

            int res = arr[from];
            for(int i = from+1; i <= to; ++i) {
                res ^= arr[i];
            }
            ans.push_back(res);
        }

        return ans;
    }
};

File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.go
func xorQueries(arr []int, queries [][]int) []int {
    ans := make([]int, len(queries))
    table := make([]int, len(arr))
    table[0] = arr[0]

    for i := 1; i < len(arr); i++ {
        table[i] = table[i-1] ^ arr[i]
    }

    for i, query := range queries {
        from := query[0]
        to := query[1]

        if from == 0 {
            ans[i] = table[to]
        } else {
            ans[i] = table[to] ^ table[from-1]
        }
    }

    return ans
}